webapp/src/slices/rsuSlice.js

```diff
@@ -1,541 +1,613 @@
-  ssmDisplay: false,
-  srmSsmList: [],
-  selectedSrm: [],
-};
+    selectedRsu: null,
+    rsuData: [],
+    rsuOnlineStatus: {},
+    rsuCounts: {},
+    countList: [],
+    currentSort: '',
+    startDate: '',
+    endDate: '',
+    messageLoading: false,
+    warningMessage: false,
+    msgType: 'BSM',
+    rsuMapData: {},
+    mapList: [],
+    mapDate: '',
+    displayMap: false,
+    bsmStart: '',
+    bsmEnd: '',
+    addPoint: false,
+    bsmCoordinates: [],
+    bsmData: [],
+    bsmDateError: false,
+    bsmFilter: false,
+    bsmFilterStep: 30,
+    bsmFilterOffset: 0,
+    issScmsStatusData: {},
+    ssmDisplay: false,
+    srmSsmList: [],
+    selectedSrm: [],
+    heatMapData: {
+        type: 'FeatureCollection',
+        features: [],
+    },
+}
 
 export const updateMessageType = (messageType) => async (dispatch) => {
-  dispatch(changeMessageType(messageType));
-  dispatch(updateRowData({ message: messageType }));
-};
+    dispatch(changeMessageType(messageType))
+    dispatch(updateRowData({ message: messageType }))
+}
 
 export const getRsuData = createAsyncThunk(
-  "rsu/getRsuData",
-  async (_, { getState, dispatch }) => {
-    const currentState = getState();
-    const token = selectToken(currentState);
-    const organization = selectOrganizationName(currentState);
-
-    await Promise.all([
-      dispatch(_getRsuInfo({ token, organization })),
-      dispatch(
-        _getRsuOnlineStatus({
-          token,
-          organization,
-          rsuOnlineStatusState: currentState.rsu.value.rsuOnlineStatus,
-        })
-      ),
-      dispatch(_getRsuCounts({ token, organization })),
-      dispatch(
-        _getRsuMapInfo({
-          token,
-          organization,
-          startDate: currentState.rsu.value.startDate,
-          endDate: currentState.rsu.value.endDate,
-        })
-      ),
-      dispatch(_getSsmSrmData({ token })),
-    ]);
-
-    return;
-  },
-  {
-    condition: (_, { getState }) => selectToken(getState()),
-  }
-);
+    'rsu/getRsuData',
+    async (_, { getState, dispatch }) => {
+        const currentState = getState()
+        const token = selectToken(currentState)
+        const organization = selectOrganizationName(currentState)
+
+        await Promise.all([
+            dispatch(_getRsuInfo({ token, organization })),
+            dispatch(
+                _getRsuOnlineStatus({
+                    token,
+                    organization,
+                    rsuOnlineStatusState:
+                        currentState.rsu.value.rsuOnlineStatus,
+                })
+            ),
+            dispatch(_getRsuCounts({ token, organization })),
+            dispatch(
+                _getRsuMapInfo({
+                    token,
+                    organization,
+                    startDate: currentState.rsu.value.startDate,
+                    endDate: currentState.rsu.value.endDate,
+                })
+            ),
+            // dispatch(_getSsmSrmData({ token })),
```

Comment:
Good catch. I noticed this function was causing the initial load for the webapp to take much longer than it should be. I have now moved this to be called directly instead of a part of the startup getRsuData function. It only runs when a MAP is being visualized instead of before.